pandas/core/indexes/base.py

```diff
@@ -4275,8 +4272,7 @@ def _ensure_index(index_like, copy=False):
 def _get_na_value(dtype):
     if is_datetime64_any_dtype(dtype) or is_timedelta64_dtype(dtype):
         return libts.NaT
-    return {np.datetime64: libts.NaT,
```

Comment:
if you would make a note, we already have routines for this entire bizness of ``_get_na_value``

``pandas.core.dtypes.missing.infer_dtype_for_scalar``, so these should be replaced